clap-utils/src/input_validators.rs

```diff
@@ -148,6 +148,33 @@ where
     }
 }
 
+pub fn is_url_or_moniker<T>(string: T) -> Result<(), String>
+where
+    T: AsRef<str> + Display,
+{
+    match url::Url::parse(&normalize_to_url_if_moniker(string.as_ref())) {
+        Ok(url) => {
+            if url.has_host() {
+                Ok(())
+            } else {
+                Err("no host provided".to_string())
+            }
+        }
+        Err(err) => Err(format!("{}", err)),
+    }
+}
+
+pub fn normalize_to_url_if_moniker(url_or_moniker: &str) -> String {
+    match url_or_moniker {
```

Comment:
Nice!  I've been wanting to do this for awhile as well.  WDYT about adding (undocumented) short synonyms, "m", "t", "d", "l"?

Reply:
nice idea: https://github.com/solana-labs/solana/pull/14409/commits/7f880aafa84d06803ae1756d1fe9cf3cd261d4d4